### DISCUSSION TITLE
Dock Action Not Working

### DISCUSSION THREAD

**Discussion by zonyl** [iRobotEducation/create3_docs#240] [Questions] [answered]
I am able to successfully send an "undock" message to the robot, however, "dock" does not work.   It just hangs looking for action server.  I have tried both fast and cyclone with the same results.

```
jason@ros1:~$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"
jason@ros1:~$ rosversion -d
humble
jason@ros1:~$ ros2 action list -t
/audio_note_sequence [irobot_create_msgs/action/AudioNoteSequence]
/dock [irobot_create_msgs/action/DockServo]
/drive_arc [irobot_create_msgs/action/DriveArc]
/drive_distance [irobot_create_msgs/action/DriveDistance]
/led_animation [irobot_create_msgs/action/LedAnimation]
/navigate_to_position [irobot_create_msgs/action/NavigateToPosition]
/rotate_angle [irobot_create_msgs/action/RotateAngle]
/undock [irobot_create_msgs/action/Undock]
jason@ros1:~$ ros2 action send_goal /dock irobot_create_msgs/action/Dock "{}"
Waiting for an action server to become available...
```

**Comment by alsora** [comment, marked as answer]
Hi @zonyl,

You can see from the action list output that the dock action type is named `DockServo` and not `Dock`.
Please try again with the correct name for the type and it should work. 

An additional note: your system is using ROS 2 Humble, but it looks like your robot is using ROS 2 Galactic. 
It's not recommended to mix different versions of ROS. 
We currently have a beta release of the create 3 firmware for Humble (find it here https://iroboteducation.github.io/create3_docs/releases/h_0_0/)
Among the various updates, in this release we also renamed the dock action to just `Dock` as the previous name was a source of confusion for the users.

**Reply by alsora** [reply]
FOR FUTURE READERS: the workaround mentioned by @slowrunner in the messages above is not correct and it should not be followed.

You shouldn't have cloned the `irobot_create_msgs` repository in the first place.
If you do that, you should make sure that you are cloning/building the correct version as indicated in the firmware page of the Create 3 build that you are using https://iroboteducation.github.io/create3_docs/releases/g_3_1/.

Please look at https://github.com/iRobotEducation/create3_docs/discussions/251 for more details.

**Comment by zonyl** [comment]
Updating the firmware to the beta of galactic worked and fixed my problem.
Thank you!!
